maskededittext/src/main/java/com/hipo/maskededittext/maskers/CurrencyMasker.kt

```diff
@@ -25,11 +29,17 @@ class CurrencyMasker(override val mask: Mask, override val onTextMaskedListener:
         if (numList.first().isEmpty()) {
             numList[0] = "0"
         }
-        val formattedTextBuilder = StringBuilder(mask.maskPattern)
-            .append(NumberFormat.getNumberInstance(Locale.US).format(numList.first().toFloat()))
-        if (numList.size > 1) {
-            formattedTextBuilder.append(POINT)
-                .append(numList[1].substring(0, if (numList[1].length < 3) numList[1].length else 2))
+
+        val formattedTextBuilder = StringBuilder(mask.maskPattern).apply {
+            append(NumberFormat.getNumberInstance(Locale.US).format(numList.first().toFloat()))
+            if (numList.size > 1) {
+                append(POINT)
+                if (currencyDecimalLimit == NO_DECIMAL_LIMIT) {
+                    append(numList[1])
+                } else {
+                    append(getLimitedDecimalString(numList[1]))
+                }
+            }
```

Comment:
Is it possible to do that with stringBuilder? would be nice experiment.

Reply:
Unfortunately, we can't give init param (mask.maskPattern) to buildString.